### DISCUSSION TITLE
How to keep winget itself up to date

### DISCUSSION THREAD

**Discussion by ToVolkmann** [microsoft/winget-cli#4100] [Configuration]
Hello everybody,
after stumble on the problem with Win10 23H2 ( https://github.com/microsoft/winget-cli/discussions/3964 ) I'm wondering how to keep winget itself up to date. I wasn't able to find the documentation how winget itself will recieve updates or how they can be triggered. Is there a "best practice"? Maybe or especially in combination with Intune?

Thanks in advance

**Comment by Trenly** [comment]
If you install AppInstaller through the Microsoft Store, the store will automatically keep it up to date. Alternatively, if you’re using version 1.6 you should be able to periodically run `winget install appinstaller` to update winget

**Reply by ToVolkmann** [reply]
Thank you for the answer. Would you specify which store you mean? MS Business store (old / new), MS public store?

**Reply by mdanish-kh** [reply]
Just a slight correction to the above suggestion. `winget install appinstaller` doesn't work. You can instead use `winget upgrade Microsoft.AppInstaller` or simply `winget upgrade winget`.  (as suggested above, this will only work if you are on winget 1.6 or above, and will only upgrade to a stable version)

Regarding MS Store, this is the link to the package in the store: https://www.microsoft.com/p/app-installer/9nblggh4nns1

You can also get the latest stable release through this link: https://aka.ms/getwinget
Latest preview release is available through this link: https://aka.ms/getwingetpreview

**Reply by ToVolkmann** [reply]
Thank you for the clarification!

**Comment by nathanpovo** [comment]
For those with a version of `winget` older than `1.6`, you can use the following command to install the latest release:

```powershell
Add-AppxPackage -Path "https://github.com/microsoft/winget-cli/releases/latest/download/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle" -ForceApplicationShutdown
```

Taken from https://github.com/at-wr/blog/discussions/3#discussioncomment-9820754

**Reply by schittli** [reply]
> For those with a version of `winget` older than `1.6`, you can use the following command to install the latest release:
> 
> ```powershell
> Add-AppxPackage -Path "https://github.com/microsoft/winget-cli/releases/latest/download/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle" -ForceApplicationShutdown
> ```
> 
> Taken from [at-wr/blog#3 (reply in thread)](https://github.com/at-wr/blog/discussions/3#discussioncomment-9820754)

As usual, it's Microsoft. Pure greed, broken functions, and stupid designs:
1. This workaround does not work. Of course.
2. We can lend you our second-year apprentices. They create concepts and program solutions in which the 5 most important use cases work rock-solid. and upgrading 

And the fact that a tool can update itself is a top priority in order to be able to reliably correct security gaps at any time.
